roles/adoption_osp_deploy/templates/os_net_config_overcloud.yml.j2

```diff
@@ -0,0 +1,42 @@
+#jinja2: trim_blocks:True, lstrip_blocks:True
+network_config:
+- type: ovs_bridge
+  name: br-ex
+  mtu: {{ interface_mtu }}
+  use_dhcp: false
+  dns_servers:
+  {% for dns_ip in dns_server | default([]) %}
+  - {{ dns_ip }}
+  {% endfor %}
+  - {{ gateway_ip }}
+  domain: []
+  addresses:
+  - ip_netmask: {{ ctlplane_ip }}/{{ ctlplane_cidr }}
```

Comment:
I recently had intermittent ssh issues with this config, while having the ctlplane ip (that I assume we use to ssh to the node) on a dedicated nic didn't present such problem. 
I'm not asking to change this layout at this point, but just wanted to point out an issue that I faced "a lot" during [1] testing, while not present in the infrared based jobs where the ssh interface is on a dedicated nic.

[1] https://review.rdoproject.org/r/c/testproject/+/53696

Reply:
@fmount tbh at first I thought this would pose the problems you mentioned, but it seems to work fine in the machine I'm currently testing on. How does the config look like when you have the controlplane ip in a dedicated nic? What ip do you assign to the bridge?

Reply:
I have to check how the downstream env looks like, I just faced that issue during the ceph migration playbook, when you need to run `os-net-config` to update the nodes' net config.
The interesting part is that I'm not facing it anymore in my last attempts, so I'm confused about this one.
I think downstream we simply do not have a `ovs_bridge` for the ssh net nic.
Example:

```
1: lo    inet 127.0.0.1/8 scope host lo\       valid_lft forever preferred_lft forever
2: enp1s0    inet 192.168.24.8/24 brd 192.168.24.255 scope global enp1s0\       valid_lft forever preferred_lft forever
2: enp1s0    inet 192.168.24.54/32 brd 192.168.24.255 scope global enp1s0\       valid_lft forever preferred_lft forever
7: br-ex    inet 10.0.0.119/24 brd 10.0.0.255 scope global br-ex\       valid_lft forever preferred_lft forever
8: vlan30    inet 172.17.3.109/24 brd 172.17.3.255 scope global vlan30\       valid_lft forever preferred_lft forever
8: vlan30    inet 172.17.3.135/32 brd 172.17.3.255 scope global vlan30\       valid_lft forever preferred_lft forever
9: vlan70    inet 172.17.5.142/24 brd 172.17.5.255 scope global vlan70\       valid_lft forever preferred_lft forever
9: vlan70    inet 172.17.5.63/32 brd 172.17.5.255 scope global vlan70\       valid_lft forever preferred_lft forever
10: vlan50    inet 172.17.2.69/24 brd 172.17.2.255 scope global vlan50\       valid_lft forever preferred_lft forever
11: vlan20    inet 172.17.1.119/24 brd 172.17.1.255 scope global vlan20\       valid_lft forever preferred_lft forever
12: vlan40    inet 172.17.4.131/24 brd 172.17.4.255 scope global vlan40\       valid_lft forever preferred_lft forever
```
and

```
---
network_config:
- type: interface
  name: nic1
  use_dhcp: false
  dns_servers: ['172.16.0.1', '10.0.0.1']
  domain: []
  addresses:
  - ip_netmask: 192.168.24.8/24
  routes: [{'default': True, 'nexthop': '192.168.24.1'}]
- type: vlan
  vlan_id: 20
  device: nic1
  addresses:
  - ip_netmask: 172.17.1.119/24
  routes: []
- type: vlan
  vlan_id: 40
  device: nic1
  addresses:
  - ip_netmask: 172.17.4.131/24
  routes: []
- type: ovs_bridge
  name: br-tenant
  use_dhcp: false
  members:
  - type: interface
    name: nic2
    primary: true
  - type: vlan
    vlan_id: 50
    addresses:
    - ip_netmask: 172.17.2.69/24
    routes: []
  - type: vlan
    vlan_id: 30
    addresses:
    - ip_netmask: 172.17.3.109/24
    routes: []
  - type: vlan
    vlan_id: 70
    addresses:
    - ip_netmask: 172.17.5.142/24
    routes: []
- type: ovs_bridge
  name: br-ex
  dns_servers: ['172.16.0.1', '10.0.0.1']
  domain: []
  use_dhcp: false
  addresses:
  - ip_netmask: 10.0.0.119/24
  routes: []
  members:
  - type: interface
    name: nic3
    primary: true
```

Not sure it helps and you don't have to change what you have because of this, but hope this gives you the idea.

Reply:
thanks that's helpful. It's quite different to what I have and tbh I'm not sure how much would I need to change on the tripleo side to use something like this, for now I'm leaning on keeping the current one and having this discussion here for reference in case we start seeing problems with the interface. Now I wonder though, is there some topology where the current setup will be problematic and will need to be changed? Should we have this template as part of the scenario definition?

Reply:
I'm not sure, maybe there's not, so I think we can continue with the current implementation and double check or provide overrides later if necessary